packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts

```diff
@@ -22,88 +22,67 @@ export const STABLE_ATTRIBUTES = [
-  // if it thinks the part is an identifier. The condition it uses is to checks whether a digit is
-  // present.
-  //
-  // Here, we use the same strategy: if a the value contains a digit, we consider it generated. This
-  // strategy might be a bit naive and fail in some cases, but there are many fallbacks to generate
-  // CSS selectors so it should be fine most of the time. We might want to allow customers to
-  // provide their own `isGeneratedValue` at some point.
-  return /[0-9]/.test(value)
-}
+type SelectorGetter = (element: Element, actionNameAttribute: string | undefined) => string | undefined
+
+// Selectors to use if they target a single element on the whole document. Those selectors are
+// considered as "stable" and uniquely identify an element regardless of the page state. If we find
+// one, we should consider the selector "complete" and stop iterating over ancestors.
+const GLOBALLY_UNIQUE_SELECTOR_GETTERS: SelectorGetter[] = [getStableAttributeSelector, getIDSelector]
+
+// Selectors to use if they target a single element among an element descendants. Those selectors
+// are more brittle than "globally unique" selectors and should be combined with ancestor selectors
+// to improve specificity.
+const UNIQUE_AMONG_CHILDREN_SELECTOR_GETTERS: SelectorGetter[] = [
+  getStableAttributeSelector,
+  getClassSelector,
+  getTagNameSelector,
+]
 
-function getSelectorFromElement(
-  targetElement: Element,
-  globallyUniqueSelectorStrategies: GetSelector[],
-  uniqueAmongChildrenSelectorStrategies: GetSelector[],
-  { useCombinedSelectors = false } = {}
-): string {
+export function getSelectorFromElement(targetElement: Element, actionNameAttribute: string | undefined) {
   let targetElementSelector = ''
   let element: Element | null = targetElement
 
   while (element && element.nodeName !== 'HTML') {
     const globallyUniqueSelector = findSelector(
       element,
-      globallyUniqueSelectorStrategies,
+      GLOBALLY_UNIQUE_SELECTOR_GETTERS,
       isSelectorUniqueGlobally,
-      useCombinedSelectors ? targetElementSelector : undefined
+      actionNameAttribute,
+      targetElementSelector
     )
     if (globallyUniqueSelector) {
-      return combineSelector(globallyUniqueSelector, targetElementSelector)
+      return globallyUniqueSelector
     }
 
     const uniqueSelectorAmongChildren = findSelector(
       element,
-      uniqueAmongChildrenSelectorStrategies,
+      UNIQUE_AMONG_CHILDREN_SELECTOR_GETTERS,
       isSelectorUniqueAmongSiblings,
-      useCombinedSelectors ? targetElementSelector : undefined
-    )
-    targetElementSelector = combineSelector(
-      uniqueSelectorAmongChildren || getPositionSelector(element) || getTagNameSelector(element),
+      actionNameAttribute,
       targetElementSelector
     )
+    targetElementSelector =
+      uniqueSelectorAmongChildren ||
+      combineSelector(getPositionSelector(element) || getTagNameSelector(element), targetElementSelector)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Would it work like so:
```suggestion
      combineSelector(getPositionSelector(element), targetElementSelector)
```
It passes tests

Reply:
Hum that's right, it doesn't seem necessary. Let me improve the situation

Reply:
Done!